tests/ballerina-test-integration/src/test/resources/websocket/ping_pong_and_client_custom_header.bal

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/io;
+import ballerina/log;
+
+@final string REMOTE_BACKEND_URL = "ws://localhost:15500/websocket";
+@final string ASSOCIATED_CONNECTION = "ASSOCIATED_CONNECTION";
+@final string data = "data";
+@final blob APPLICATION_DATA = data.toBlob("UTF-8");
+
+@http:WebSocketServiceConfig {
+    path: "/pingpong/ws"
+}
+service<http:WebSocketService> PingPongTestService bind { port: 9090 } {
```

Comment:
IMO we should separate this into two separate files since they are used for totally different tests as I mentioned above.